app/services/catalog_remediator.rb

```diff
@@ -49,20 +53,21 @@ def zipped_moab_versions_beyond_expiry
 
   def zipped_moab_versions_with_errors
     zipped_moab_versions_beyond_expiry.to_a.filter_map do |zipped_moab_version|
-      audit_results = audit_results_for(zipped_moab_version)
-      # If this check returns false, it indicates a lack of zip parts, in which
-      # case we still want to delete the ZMV, so short-circuit the rest of the
-      # block
+      audit_results = empty_audit_results(zipped_moab_version)
+      # when Audit::CatalogToArchive.check_child_zip_part_attributes() returns true, we have all ZipParts for the
+      #   ZippedMoabVersion, so continue to the next check
       next [zipped_moab_version, audit_results] unless check_child_zip_part_attributes(zipped_moab_version, audit_results)
 
+      # S3Audit.check_replicated_zipped_moab_version creates audit_results.error_results if problem with existence or
+      #   checksum of replicated files
```

Comment:
I do like surfacing that this method mutates the audit results, which better connects it to the following line.